app/test/vesting_test.go

```diff
@@ -0,0 +1,674 @@
+package app_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/app/encoding"
+	"github.com/celestiaorg/celestia-app/test/util/testfactory"
+	"github.com/celestiaorg/celestia-app/test/util/testnode"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	vestingtypes "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	distributiontypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	tmtime "github.com/tendermint/tendermint/types/time"
+)
+
+const (
+	totalAccountsPerType = 300
+	initBalanceForGasFee = 10
+	vestingAmount        = testfactory.BaseAccountDefaultBalance
+)
+
+type accountDispenser struct {
+	names   []string
+	counter int
+}
+
+type accountType int
+
+const (
+	RegularAccountType accountType = iota + 1
+	DelayedVestingAccountType
+	PeriodicVestingAccountType
+	ContinuousVestingAccountType
+)
```

Comment:
[optional] since this test file is so large, I wonder if it would make sense to have one file per type of account under test

Reply:
On second thought, there is so much similar code across tests that it may not make sense to split them up

Reply:
now we have shrank the code a lot, so it is considered done ;)